chirp/drivers/mml_jc8810.py

```diff
@@ -0,0 +1,1070 @@
+TOT_LIST = ["Off"] + ["%s seconds" % x for x in range(30, 270, 30)]
+VOX_LIST = ["Off"] + ["%s" % x for x in range(1, 10)]
+VOXD_LIST = ["%s seconds" % str(x / 10) for x in range(5, 21)]
+WORKMODE_LIST = ["VFO Mode", "Channel Mode"]
+
+SKEY_CHOICES = ["FM Radio", "Flashlight", "TX Power Level",
+                "NOAA Weather", "Scan", "Search"]
+SKEY_VALUES = [0x07, 0x08, 0x0A, 0x0C, 0x1C, 0x1D]
+
+SKEY2SP_CHOICES = ["PTT B"] + SKEY_CHOICES
+SKEY2SP_VALUES = [0x01] + SKEY_VALUES
+
+SETTING_LISTS = {
+    "abr": ABR_LIST,
+    "almod": ALMODE_LIST,
+    "autolk": AUTOLK_LIST,
+    "dtmfoff": DTMFSPEED_LIST,
+    "dtmfon": DTMFSPEED_LIST,
+    "dtmfst": DTMFST_LIST,
+    "dualtx": DUALTX_LIST,
+    "encrypt": ENCRYPT_LIST,
+    "language": LANGUAGE_LIST,
+    "mdfa": MDF_LIST,
+    "mdfb": MDF_LIST,
+    "menuquit": MENUQUIT_LIST,
+    "ponmsg": PONMSG_LIST,
+    "pttid": PTTID_LIST,
+    "pttlt": PTTLT_LIST,
+    "qtsave": QTSAVE_LIST,
+    "rpste": RPSTE_LIST,
+    "rptrl": RPSTE_LIST,
+    "rxendtail": TONERXEND_LIST,
+    "save": SAVE_LIST,
+    "scode": PTTIDCODE_LIST,
+    "screv": SCREV_LIST,
+    "tailcode": TAILCODE_LIST,
+    "tone": TONE_LIST,
+    "tot": TOT_LIST,
+    "vox": OFF1TO9_LIST,
+    "voxd": VOXD_LIST,
+    "workmode": WORKMODE_LIST
+    }
+
+
+def _enter_programming_mode(radio):
+    serial = radio.pipe
+
+    exito = False
+    for i in range(0, 5):
+        serial.write(radio._magic)
+        ack = serial.read(1)
+
+        try:
+            if ack == CMD_ACK:
+                exito = True
+                break
+        except errors.RadioError:
+            LOG.debug("Attempt #%s, failed, trying again" % i)
+            pass
+
+    # check if we had EXITO
+    if exito is False:
+        msg = "The radio did not accept program mode after five tries.\n"
+        msg += "Check you interface cable and power cycle your radio."
+        raise errors.RadioError(msg)
+
+    try:
+        serial.write(b"F")
+        ident = serial.read(8)
+    except errors.RadioError:
+        raise errors.RadioError("Error communicating with radio")
+
+    if ident not in radio._fingerprint:
+        LOG.debug(util.hexprint(ident))
+        raise errors.RadioError("Radio returned unknown identification string")
+
+
+def _exit_programming_mode(radio):
+    serial = radio.pipe
+    try:
+        serial.write(b"E")
+    except errors.RadioError:
+        raise errors.RadioError("Radio refused to exit programming mode")
+
+
+def _read_block(radio, block_addr, block_size):
+    serial = radio.pipe
+
+    cmd = struct.pack(">cHb", b'R', block_addr, block_size)
+    expectedresponse = b"R" + cmd[1:]
+    LOG.debug("Reading block %04x..." % (block_addr))
+
+    try:
+        serial.write(cmd)
+        response = serial.read(4 + block_size)
+        if response[:4] != expectedresponse:
+            raise Exception("Error reading block %04x." % (block_addr))
+
+        block_data = response[4:]
+    except errors.RadioError:
```

Comment:
And this. I'm guessing maybe this pattern was copied from something that wrapped the reads and writes in another function that *could* raise that error? In this case, you probably want to `raise errors.RadioError` on L308, catch that specifically, re-`raise` it, and then catch `Exception` to grab anything else and raise an `errors.RadioError`.